packages/surveys/src/utils/results-availability/has-user-responded.ts

```diff
@@ -0,0 +1,25 @@
+import { queryFeatures, IQueryResponse } from "@esri/arcgis-rest-feature-layer";
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { IStakeholderItem } from "../../types";
+
+/**
+ * check is provided user has any survey submissions
+ * @param {IStakeholderItem.url} _url feature service url
+ * @param {string} username the username to check for survey responses
+ * @param {IRequestOptions} requestOptions The request options
+ * @returns {Promise<boolean>}
+ */
+export const hasUserResponded = (
+  _url: IStakeholderItem["url"],
+  username: string,
+  requestOptions: IRequestOptions
+): Promise<boolean> => {
+  const url = `${_url}/0`;
```

Comment:
is there a chance that the URL already includes the layer index? Should you check if it does before appending (otherwise it will get the first individual Feature, so still be a 200 OK)

Reply:
The url should come from a feature service item, which I believe does not include the layer index.

Reply:
we do a lot of CYA in URL building in the rest of the library.